Misc test stubs types

UnknownPlatypus opened the pull request:
Some more misc test stubs update

UnknownPlatypus committed:
Django test init file errors

UnknownPlatypus committed:
Add other lazy regex to `allowlist.txt`

UnknownPlatypus committed:
Fix some `django.test.client` types

UnknownPlatypus committed:
Misc `django.text.html` cleanups

sobolevn commented on django-stubs/test/html.pyi:
```diff
@@ -1,12 +1,15 @@
 from typing_extensions import TypeAlias
 
-WHITESPACE: Any
+ASCII_WHITESPACE: Pattern
```
```suggestion
ASCII_WHITESPACE: Pattern[str]
```

UnknownPlatypus replied:
Thanks, I've updated other `Pattern` which I believe also miss this.

While at it, I also updated the `allowlist.txt` to reference every regex pattern in the `Lazy regex` section. 
I found some missing running `rg ": Pattern" -N | cut -d":" -f2 | xargs -I {} rg {} ./scripts/stubtest/allowlist_todo.txt`

sobolevn reviewed (commented):
Thanks!

UnknownPlatypus committed:
Replace `Pattern` with `Pattern[str]`

UnknownPlatypus committed:
Update allowlist_todo to move all regex pattern to allowlist

Using `rg ": Pattern" -N | cut -d":" -f2 | xargs -I {} rg {} ./scripts/stubtest/allowlist_todo.txt`

sobolevn reviewed (approved):
Thanks!
